src/OSPSuite.Infrastructure.Import/Core/Mappers/DataSetToDataRepositoryMapper.cs

```diff
@@ -34,9 +49,10 @@ public DataRepository ConvertImportDataSet(ImportedDataSet dataSet)
             dataRepository.ExtendedProperties.Add(new ExtendedProperty<string>() { Name = metaDataDescription.Name, Value = metaDataDescription.Value });
          }
 
+         var warningFlag = false;
          foreach (var column in dataSet.ParsedDataSet.Data)
          {
-            convertParsedDataColumn(dataRepository, column, dataSet.FileName);
+            warningFlag |= convertParsedDataColumnAndReturnWarningFlag(dataRepository, column, dataSet.FileName);
```

Comment:
Isn't this a case for a any?
```
var warningFlag = dataSet.ParsedDataSet.Data.any(x=>convertParsedDataColumnAndReturnWarningFlag(dataRepository, x, dataSet.FileName))
```

Reply:
Yes, it was already on a foreach and I did not really realized it... thanks!!

Reply:
Actually, same as below here. If I use any, it will return as soon as it finds some true. Any is not intended for changing the data. Normally foreach should not translate to any. Don't you agree?

Reply:
yeah. The problem is that your method has SIDE EFFECTS and is not void. That's why it through me off.
I am ok with this for now because we are running out of time
But side effects method should not return values. If we make our functions as pure as possible, there is no surprises as to what the code does and one does not need to read the details of each function